openmdao/components/tests/test_implicit_func_comp.py

```diff
@@ -0,0 +1,233 @@
+
+        x_=np.array([1, 2, -3])
+        A = np.array([[5.0, -3.0, 2.0], [1.0, 7.0, -4.0], [1.0, 0.0, 8.0]])
+        b = A.dot(x_)
+
+        def resid_func(A, b, x):
+            rx = A.dot(x) - b
+            return rx
+
+        f = (omf.wrap(resid_func)
+                .add_input('A', shape=A.shape)
+                .add_input('b', shape=b.shape)
+                .add_output('x', resid='rx', val=x_))
+
+        prob = om.Problem()
+        model = prob.model
+
+        model.add_subsystem('p1', om.IndepVarComp('A', A))
+        model.add_subsystem('p2', om.IndepVarComp('b', b))
+
+        lingrp = model.add_subsystem('lingrp', om.Group(), promotes=['*'])
+        lingrp.add_subsystem('lin', om.ImplicitFuncComp(f))
+
+        model.connect('p1.A', 'lin.A')
+        model.connect('p2.b', 'lin.b')
+
+        prob.setup()
+
+        lingrp.linear_solver = om.ScipyKrylov()
+
+        prob.set_solver_print(level=0)
+        prob.run_model()
+
+        assert_near_equal(prob['lin.x'], x_, .0001)
+        assert_near_equal(prob.model._residuals.get_norm(), 0.0, 1e-10)
+
+        model.run_apply_nonlinear()
+
+        with model._scaled_context_all():
+            val = model.lingrp.lin._residuals['x']
+            assert_near_equal(val, np.zeros((3, )), tolerance=1e-8)
+
+    def test_component_model_w_linearize(self):
+
+        def apply_nonlinear(z, x, y1, y2, R_y1, R_y2):
+
+            z0 = z[0]
+            z1 = z[1]
+
+            return 0, (y1**.5 + z0 + z1) - y2, (z0**2 + z1 + x - 0.2*y2) - y1 - R_y1, (y1**.5 + z0 + z1) - y2 - R_y2
+
+        f = (omf.wrap(apply_nonlinear)
+                .add_input('z', val=np.array([-1., -1.]))
+                .add_input('x', val=2.)
+                .add_output('y1', resid='r_y1')
+                .add_output('y2', resid='r_y2')
+                .add_output('R_y1', resid='r_R_y1')
+                .add_output('R_y2', resid='r_R_y1')
+                .declare_partials('y1', 'y1')
+                .declare_partials('y2', ['z', 'y1', 'y2'])
+                .declare_partials('R_y1', ['R_y1', 'x', 'z', 'y1', 'y2'])
+                .declare_partials('R_y2', ['R_y2','z', 'y1', 'y2']))
+
+        def linearize(z, x, y1, y2, R_y1, R_y2, J):
+
+            J['y1', 'y1'] = -1.
+            J['R_y1','R_y1'] = -1
+
+            J['R_y1', 'x'] = [1]
+            J['R_y1', 'z'] = [2*z[0], 1]
+            J['R_y1', 'y1'] = -1.
+            J['R_y1', 'y2'] = -0.2
+
+            J['y2','y2'] = -1
+
+            J['R_y2','R_y2'] = -1
+            J['y2','z'] = [1, 1]
+            J['y2','y1'] = 0.5*y1**-0.5
+
+            J['R_y2','y2'] = -1
+            J['R_y2','z'] = [1, 1]
+            J['R_y2','y1'] = 0.5*y1**-0.5
+
+        p_opt = om.Problem()
+
+        p_opt.model = om.ImplicitFuncComp(f, linearize=linearize) #, solve_nonlinear=solve_nonlinear)
+
+        newton = p_opt.model.nonlinear_solver = om.NewtonSolver(solve_subsystems=False)
+        newton.options['iprint'] = 0
```

Comment:
couldn't you do this in one line by passing `iprint` as an arg as well?
`p_opt.model.nonlinear_solver = om.NewtonSolver(solve_subsystems=False, iprint=0)`

Reply:
fixed